vault/expiration.go

```diff
@@ -1812,6 +1822,10 @@ func (m *ExpirationManager) updatePendingInternal(le *leaseEntry) {
 
 		m.removeFromPending(m.quitContext, le.LeaseID, false)
 		m.irrevocable.Store(le.LeaseID, m.inMemoryLeaseInfo(le))
+
+		if !leaseInIrrevocable {
```

Comment:
might be overly-cautious, but it might be good to note that we care about whether the lease was in the irrevocable map *before* adding it a couple lines above. This could be done either with a comment, or by calling this code just before line 1824

Reply:
i think it probably makes sense to have the increment here, after the store succeeds, will add a comment to clarify!

Reply:
accfda0